### PR TITLE
N°6086 - CSV import: Treat first line as a header

### PR DESCRIPTION
**Pull request by bdalsass**

### CODE REVIEW

**Review by Molkobain — CHANGES_REQUESTED** [review]
Rather include the column number in the dict. entry and use `\Dict::Format()`, that way both the symbol (`"#"` is anglo-saxon oriented, in latin languages we might use something else) and the placement (for those who want to put the number first)

File: pages/ajax.csvimport.php [hunk 1]
```diff
@@ -227,37 +227,47 @@ function GetMappingForField($sClassName, $sFieldName, $iFieldIndex, $bAdvancedMo
...
 				$aTableData = [];
-				foreach ($aData as $aRow) {
-					$sCSSClass = 'csv_row'.($index % 2);
-					if (($bFirstLineAsHeader) && ($index == 1)) {
-						$aColumns[] = ["label" => sprintf($sFormat, $index)];
-						foreach ($aRow as $sCell) {
-							$aColumns[] = ["label" => utils::EscapeHtml($sCell)];
-						}
+
+				// iterate throw data elements...
```
Comment (piRGoif) [inline pages/ajax.csvimport.php:233]:
```suggestion
```

File: pages/ajax.csvimport.php [hunk 2]
```diff
@@ -227,37 +227,47 @@ function GetMappingForField($sClassName, $sFieldName, $iFieldIndex, $bAdvancedMo
...
+				for ($i = 0 ; $i < count($aData) ; $i++) {
+
+					// get data element
```
Comment (piRGoif) [inline pages/ajax.csvimport.php:236]:
```suggestion
```

File: pages/ajax.csvimport.php [hunk 3]
```diff
@@ -227,37 +227,47 @@ function GetMappingForField($sClassName, $sFieldName, $iFieldIndex, $bAdvancedMo
...
+					$aRow = $aData[$i];
+
+					// when first line
```
Comment (piRGoif) [inline pages/ajax.csvimport.php:239]: I would suggest removing the comment, and renaming $i to something like `$iDataLineNumber` ?

**Review by piRGoif — CHANGES_REQUESTED** [review]
Nice work thanks !
As this is a bugfix, can this be rebased to support/2.7 ? The decision on which versions the PR will be merged are made during functional review, but it's easier to base on the lowest branch

File: pages/ajax.csvimport.php [hunk 4]
```diff
@@ -253,7 +253,7 @@ function GetMappingForField($sClassName, $sFieldName, $iFieldIndex, $bAdvancedMo
 
 						// default headers
 						for ($j = 0 ; $j < count($aRow) ; $j++) {
-							$aColumns[] = ["label" => Dict::S('UI:CSVImport:Column') . ' #' . ($j + 1)];
+							$aColumns[] = ["label" => Dict::Format('UI:CSVImport:Column',$j+1)];
```
Comment (Molkobain) [inline pages/ajax.csvimport.php]:
```suggestion
							$aColumns[] = ["label" => Dict::Format('UI:CSVImport:Column', $j+1)];
```

**Review by Molkobain — APPROVED** [review]

**Comment by Molkobain** [comment]
Ok for me, mind to change the target branch to support/3.2

**Comment by Hipska** [comment]
Also, https://github.com/Combodo/iTop/blob/4e53deec9d96093df87b233249a9c071ae458d40/.github/pull_request_template.md?plain=1#L3-L5

File: pages/ajax.csvimport.php (hunk 3 above)
Comment (bdalsass) [reply pages/ajax.csvimport.php:239]: oky done

**Comment by bdalsass** [comment]
Le changement de cible m'a foutu un vrac pas permis avec conflit et plein de truc en sortie.
J'ai récupéré les commits et changé un bout de code sur la gestion du nombre max d'éléments qui n'était pas cohérent si on utilise la première ligne comme entêtes.

**Pull request closed without merging**